frontend/src/pages/ResourcesDashboard/index.js

```diff
@@ -0,0 +1,164 @@
+import React, { useMemo, useContext } from 'react';
+import PropTypes from 'prop-types';
+import { Helmet } from 'react-helmet';
+import { v4 as uuidv4 } from 'uuid';
+import { Grid, GridContainer } from '@trussworks/react-uswds';
+import FilterPanel from '../../components/filter/FilterPanel';
+import ResourceList from '../../widgets/ResourceList';
+import ResourcesDashboardOverview from '../../widgets/ResourcesDashboardOverview';
+import { getUserRegions } from '../../permissions';
+import './index.scss';
+import { formatDateRange, expandFilters } from '../../utils';
+import useSessionFiltersAndReflectInUrl from '../../hooks/useSessionFiltersAndReflectInUrl';
+import UserContext from '../../UserContext';
+import { RESOURCES_DASHBOARD_FILTER_CONFIG } from './constants';
+import RegionPermissionModal from '../../components/RegionPermissionModal';
+import { buildDefaultRegionFilters, showFilterWithMyRegions } from '../regionHelpers';
+
```

Comment:
I notice a lot of stuff copy and pasted here (it's awesome we can reuse so much of our existing work, indicates we did a pretty good job on the first one) - do you think, since we're a little ahead of the curve here we could take this opportunity to move some of these functions outside of the dashboard components and into a helper file of some kind? Not a huge priority, but it feels like a little alarm going off when we are copying and pasting so much stuff around exactly. 

(It will also help with code coverage thresholds)

Reply:
I agree its a good idea to create components for reusability. My main concern here is that this page will see a major overhaul once it gets designed out by UX. 

I guess what exactly did you see about moving into separate components?

We should have some code coverage for this new code with existing tests.

Reply:
```js
  const hasCentralOffice = useMemo(() => (
    user && user.homeRegionId && user.homeRegionId === 14
  ), [user]);
  const regions = useMemo(() => getUserRegions(user), [user]);
  const userHasOnlyOneRegion = useMemo(() => regions.length === 1, [regions]);
  const defaultRegion = useMemo(() => regions[0].toString(), [regions]);

  const allRegionsFilters = useMemo(() => buildDefaultRegionFilters(regions), [regions]);

  const getFiltersWithAllRegions = () => {
    const filtersWithAllRegions = [...allRegionsFilters];
    filtersWithAllRegions.push({
      id: uuidv4(),
      topic: 'startDate',
      condition: 'is within',
      query: defaultDate,
    });
    return filtersWithAllRegions;
  };

  const centralOfficeWithAllRegionFilters = getFiltersWithAllRegions();

  const defaultFilters = useMemo(() => {
    if (hasCentralOffice) {
      return centralOfficeWithAllRegionFilters;
    }

    return [
      {
        id: uuidv4(),
        topic: 'region',
        condition: 'is',
        query: defaultRegion,
      },
      {
        id: uuidv4(),
        topic: 'startDate',
        condition: 'is within',
        query: defaultDate,
      },
    ];
  }, [defaultRegion, hasCentralOffice, centralOfficeWithAllRegionFilters]);
  
    const [filters, setFilters] = useSessionFiltersAndReflectInUrl(FILTER_KEY, defaultFilters);

  // Apply filters.
  const onApplyFilters = (newFilters, addBackDefaultRegions) => {
    if (addBackDefaultRegions) {
      // We always want the regions to appear in the URL.
      setFilters([
        ...allRegionsFilters,
        ...newFilters,
      ]);
    } else {
      setFilters(newFilters);
    }
  };
  
    // Remove Filters.
  const onRemoveFilter = (id, addBackDefaultRegions) => {
    const newFilters = [...filters];
    const index = newFilters.findIndex((item) => item.id === id);
    if (index !== -1) {
      newFilters.splice(index, 1);
      if (addBackDefaultRegions) {
        // We always want the regions to appear in the URL.
        setFilters([...allRegionsFilters, ...newFilters]);
      } else {
        setFilters(newFilters);
      }
    }
  };

```

What I mean is that the whole above section is exactly the same in both files, maybe we can extract some of these functions so that we aren't maintaining them in two places